hazelcast/src/hazelcast/client/compact.cpp

```diff
@@ -0,0 +1,678 @@
+    if (field_descriptor == fields.end()) {
+        BOOST_THROW_EXCEPTION(unknown_field_exception(field_name));
+    }
+    return field_descriptor->second;
+}
+
+const pimpl::field_descriptor&
+compact_reader::get_field_descriptor(const std::string& field_name,
+                                     enum pimpl::field_kind kind) const
+{
+    const auto& field_descriptor = get_field_descriptor(field_name);
+    if (field_descriptor.field_kind != kind) {
+        BOOST_THROW_EXCEPTION(
+          unexpected_field_kind(field_descriptor.field_kind, field_name));
+    }
+    return field_descriptor;
+}
+
+exception::hazelcast_serialization
+compact_reader::unknown_field_exception(const std::string& field_name) const
+{
+    return { "compact_reader",
+             (boost::format("Unknown field name %1% on %2% ") % field_name %
+              schema)
+               .str() };
+}
+
+exception::hazelcast_serialization
+compact_reader::unexpected_field_kind(enum pimpl::field_kind field_kind,
+                                      const std::string& field_name) const
+{
+    return { "compact_reader",
+             (boost::format("Unexpected fieldKind %1% for %2% on %3%") %
+              field_kind % field_name % schema)
+               .str() };
+}
+
+exception::hazelcast_serialization
+compact_reader::unexpected_null_value(const std::string& field_name,
+                                      const std::string& method_suffix)
+{
+    return { "compact_reader",
+             (boost::format(
+                "Error while reading %1%. null value can not be read via "
+                "get_%2% methods. Use get_nullable_%2%  instead.") %
+              field_name % method_suffix)
+               .str() };
+}
+
+size_t
+compact_reader::read_fixed_size_position(
+  const pimpl::field_descriptor& field_descriptor) const
+{
+    int primitive_offset = field_descriptor.offset;
+    return primitive_offset + data_start_position;
+}
+
+size_t
+compact_reader::read_var_size_position(
+  const pimpl::field_descriptor& field_descriptor) const
+{
+    int index = field_descriptor.index;
+    int offset =
+      get_offset(object_data_input, variable_offsets_position, index);
+    return offset == pimpl::offset_reader::NULL_OFFSET
+             ? pimpl::offset_reader::NULL_OFFSET
+             : offset + data_start_position;
+}
+
+int32_t
+compact_reader::read_int32(const std::string& field_name)
+{
+    const auto& fd = get_field_descriptor(field_name);
+    const auto& fieldKind = fd.field_kind;
+    switch (fieldKind) {
+        case pimpl::field_kind::INT32:
+            return object_data_input.read<int32_t>(
+              read_fixed_size_position(fd));
+        case pimpl::field_kind::NULLABLE_INT32:
+            return get_variable_size_as_non_null<int32_t>(
+              fd, field_name, "int32");
+        default:
+            BOOST_THROW_EXCEPTION(unexpected_field_kind(fieldKind, field_name));
+            return -1;
```

Comment:
This is unreachable and removable.

Reply:
Fixed here 33186c83e8fcb88310e61706f80bbf9c8ea32528